CONTRIBUTING.md

```diff
@@ -2,9 +2,49 @@
 
 Thanks for contributing to libphonenumber!
 
-Please read the following before filing an issue. **Note that we are _not_
-accepting pull requests for validation or formatting metadata updates at this
-time.**
+Please read the following before filing an issue or sending a pull request.
+
+We hope these guidelines will enhance your experience as a contributor to our
+library and know that we appreciate the time you put into making it better.
+
+## Pull requests
+
+We do not accept pull requests for validation, formatting, or timezone metadata
+updates.
+
+For changes specific to windows builds, see [Filing a code
+issue](#filing-a-code-issue) and make sure you have found a reviewer and tester
+before sending the pull request.
+
+We are happy to review and accept pull requests for the following:
+
+*   Carrier updates
+*   Geocoding updates
+*   Bug fixes
+*   API changes
+*   Documentation and code hygiene cleanups
+
+This list is not exhaustive.
+
+To get your pull request merged, we need the following:
+
+*   A CLA signature. See below.
+*   Responsiveness to back and forth review comments, because we will do a code
+    review.
+*   A clear description of the issue the pull request intends to solve.
+*   Tests that illustrate the issue, if applicable.
+*   Willingness and ability to implement the changes in C++, Java, and JS if
+    applicable. For example, changes to `PhoneNumberUtil.java` must be ported to
+    `phonenumberutil.cc` and `phonenumberutil.js`.
+    *   The initial pull request may contain the implementation in only one
+        language to get approval on the approach.
+
+If this is not possible, please file an issue.
+
+Before spending significant time on a bug fix or API change pull request, we
+strongly encourage that you file an issue so we can discuss whether this is
+something we think should be fixed, how to do it, and whether we'd accept a pull
+request.
```

Comment:
Why? To see if others see a need for the API?
As in why not an issue?

I think the bug fix thing could also be discussed in the list but it's an additional hurdle (one has to have a Google account and be whitelisted, so they're blocked until we do that).